monitor/cron_monitor.sh

```diff
@@ -0,0 +1,60 @@
+#! /usr/bin/env bash
+# set -x
+# set -e
+
+export AWS_ACCESS_KEY_ID=$S3_ACCESS_KEY_ID
+export AWS_SECRET_ACCESS_KEY=$S3_SECRET_ACCESS_KEY
+
+if [[ -z "$S3_ENDPOINT" ]]; then
+  AWS_ARGS=""
+else
+  AWS_ARGS="--endpoint-url $S3_ENDPOINT"
+fi
+
+list_s3() {
+  if [[ -z "$S3_PREFIX" ]]; then
+    S3_PATH= "s3://$S3_BUCKET"
+  else
+    S3_PATH="s3://$S3_BUCKET/$S3_PREFIX"
+  fi
+
+  if [[ ! -z "$1" ]]; then
+    S3_PATH="$S3_PATH/$1"
+  fi
+
+  echo "Listing all files from $S3_PATH"
+  aws $AWS_ARGS s3 ls $S3_PATH/ --human-readable
+}
+s3_latest_mediawiki_dump_backup_record=$(list_s3 hourly | tail -1) 
+echo "Latest mediawiki dump record: $s3_latest_mediawiki_dump_backup_record"
+s3_latest_sql_dump_backup_record=$(list_s3 hourly | tail -2 | head -1) 
+echo "Latest sql dump record: $s3_latest_sql_dump_backup_record"
+
+t1=$(echo $s3_latest_mediawiki_dump_backup_record | awk '{print substr($0,1,19)}')
+mediawiki_dump_time=$(date -d "$t1" +%s)
+t2=$(echo $s3_latest_sql_dump_backup_record | awk '{print substr($0,1,19)}')
```

Comment:
@vijayabharathib can we use the time we got it from api response instead of from the file name?

the file created in s3 is the time we want to compare. the created time may older if its more time to dump or zip.

Reply:
getting it from api timestamp, not from file name. all good as discussed.